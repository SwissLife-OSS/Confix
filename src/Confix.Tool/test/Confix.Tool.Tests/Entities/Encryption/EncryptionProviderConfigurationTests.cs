using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;
using Confix.Entities.Component.Configuration;

namespace Confix.Tool.Tests.Entities.Encryption;

public class EncryptionProviderConfigurationTests : ParserTestBase
{
    [Fact]
    public void Parse_Should_BeValid()
    {
        ExpectValid(
            """
            {
                "type": "test"
            }
            """);
    }

    [Fact]
    public void Parse_Should_BeValid_WhenEmpty()
    {
        ExpectValid(""" { } """);
    }

    [Fact]
    public void Parse_Should_BeValid_WithEnvironmentOverrides()
    {
        ExpectValid(
            """
            {
                "type": "test",
                "environmentOverride": {
                    "test": {
                        "type": "test"
                    }
                }
            }
            """);
    }

    public override object Parse(JsonNode json)
        => EncryptionProviderConfiguration.Parse(json);
}
