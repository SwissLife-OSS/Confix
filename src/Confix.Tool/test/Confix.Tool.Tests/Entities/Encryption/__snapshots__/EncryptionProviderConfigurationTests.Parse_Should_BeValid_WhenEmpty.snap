{
  "Type": null,
  "EnvironmentOverrides": null,
  "Values": {}
}
